dissect/target/plugins/os/windows/firewall.py

```diff
@@ -0,0 +1,329 @@
+            tcpwin (string): TCP window size in bytes.
+            icmptype (string): ICMP packet type.
+            icmpcode (string): ICMP packet code.
+            info (string): Additional information.
+            path (string): Direction of the traffic, either SEND, RECEIVE, FORWARD or UNKNOWN.
+            source (path): Source path of the record log line.
+        """
+
+        for path in self.log_paths:
+            try:
+                fh = path.open("rt")
+                header = list(islice(fh, 4))
+                config = LogConfig(header)
+
+            except Exception as e:
+                self.target.log.warning("Unable to read Windows Firewall log file config in %s: %s", path, e)
+                self.target.log.debug("", exc_info=e)
+                continue
+
+            if config.version != 1.5:
+                self.target.log.warning(
+                    "Unsupported Windows Firewall log file version %r in log file %s", header[0], path
+                )
+                continue
+
+            for line in fh:
+                if not (line := line.strip()):
+                    continue
+
+                entry = FieldMap(config.fields, line.split(" ", len(config.fields)))
+                entry.ts = datetime.strptime(f"{entry.date} {entry.time}", "%Y-%m-%d %H:%M:%S").replace(
+                    tzinfo=self.target.datetime.tzinfo if config.time_format == "local" else None
+                )
+
+                del entry.date
+                del entry.time
+
+                yield WindowsFirewallLogRecord(
+                    **dict(entry),
+                    source=path,
+                    _target=self.target,
+                )
+
+
+@dataclass
+class LogConfig:
+    """Windows Firewall Log ``pfirewall.log`` file config parser.
+
+    Fields can differ depending on configuration and version.
+
+    Example structure of a regular ``pfirewall.log`` file:
+
+    .. code-block::
+
+        #Version: 1.5
+        #Software: Microsoft Windows Firewall
+        #Time Format: Local
+        #Fields: date time action protocol src-ip dst-ip src-port dst-port size tcpflags tcpsyn tcpack tcpwin icmptype icmpcode info path
+
+        2022-01-01 13:37:00 DROP UDP 1.2.3.4 5.6.7.8 1234 5678 1337 - - - - - - - RECEIVE
+    """  # noqa: E501
+
+    raw: list[str] = field(repr=False)
+    version: float | None = None
+    software: str | None = None
+    time_format: str | None = None
+    fields: list[str] | None = None
+
+    def __post_init__(self):
+        assert len(self.raw) == 4
+        _, _, self.version = self.raw[0].strip().partition("#Version: ")
+        _, _, self.software = self.raw[1].strip().partition("#Software: ")
+        _, _, self.time_format = self.raw[2].strip().lower().partition("#time format: ")
+        _, _, fields = self.raw[3].strip().partition("#Fields: ")
+        self.version = float(self.version)
+        self.fields = fields.split(" ")
+
+
+@dataclass
+class FieldMap:
```

Comment:
Is there really a benefit to using a dataclass for this? Can't you use a single dictionary comprehension with `zip`?

Reply:
I think I prefer the dictionary and zip approach since this will be a pain for typing.